Tested on RPi.

**Before this PR**:

<img width="881" alt="image" src="https://user-images.githubusercontent.com/22748355/220968664-b4082f14-564d-4f67-9c91-523c42c5b07d.png">


<details>
<summary>
<code>2023-02-23T16:22:55,636 [qtp600529253-1599] WARN  o.e.j.s.h.C.ROOT - Exception while dispatching incoming RPC call com.google.gwt.user.server.rpc.UnexpectedException: Service method 'public abstract java.util.ArrayList org.eclipse.kura.web.shared.service.GwtStatusService.getDeviceConfig(org.eclipse.kura.web.shared.model.GwtXSRFToken,boolean,boolean) throws org.eclipse.kura.web.shared.GwtKuraException' threw an unexpected exception: java.lang.ClassCastException: org.eclipse.kura.web.shared.model.GwtNetInterfaceConfig cannot be cast to org.eclipse.kura.web.shared.model.GwtWifiNetInterfaceConfig</code>
</summary>

```bash
2023-02-23T16:22:55,636 [qtp600529253-1599] WARN  o.e.j.s.h.C.ROOT - Exception while dispatching incoming RPC call
com.google.gwt.user.server.rpc.UnexpectedException: Service method 'public abstract java.util.ArrayList org.eclipse.kura.web.shared.service.GwtStatusService.getDeviceConfig(org.eclipse.kura.web.shared.model.GwtXSRFToken,boolean,boolean) throws org.eclipse.kura.web.shared.GwtKuraException' threw an unexpected exception: java.lang.ClassCastException: org.eclipse.kura.web.shared.model.GwtNetInterfaceConfig cannot be cast to org.eclipse.kura.web.shared.model.GwtWifiNetInterfaceConfig
	at com.google.gwt.user.server.rpc.RPC.encodeResponseForFailure(RPC.java:416)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:605)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
	at org.eclipse.kura.web.server.OsgiRemoteServiceServlet.processCall(OsgiRemoteServiceServlet.java:208)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
	at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
	at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at org.eclipse.kura.web.server.OsgiRemoteServiceServlet.service(OsgiRemoteServiceServlet.java:95)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.equinox.http.servlet.internal.HttpServiceRuntimeImpl$LegacyServlet.service(HttpServiceRuntimeImpl.java:1459)
	at org.eclipse.equinox.http.servlet.internal.registration.EndpointRegistration.service(EndpointRegistration.java:159)
	at org.eclipse.equinox.http.servlet.internal.servlet.ResponseStateHandler.processRequest(ResponseStateHandler.java:67)
	at org.eclipse.equinox.http.servlet.internal.context.DispatchTargets.doDispatch(DispatchTargets.java:117)
	at org.eclipse.equinox.http.servlet.internal.servlet.ProxyServlet.dispatch(ProxyServlet.java:147)
	at org.eclipse.equinox.http.servlet.internal.servlet.ProxyServlet.preprocess(ProxyServlet.java:115)
	at org.eclipse.equinox.http.servlet.internal.servlet.ProxyServlet.service(ProxyServlet.java:104)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.equinox.http.jetty.internal.HttpServerManager$InternalHttpServiceServlet.service(HttpServerManager.java:305)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:554)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:722)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:555)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:410)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:164)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.ClassCastException: org.eclipse.kura.web.shared.model.GwtNetInterfaceConfig cannot be cast to org.eclipse.kura.web.shared.model.GwtWifiNetInterfaceConfig
	at org.eclipse.kura.web.server.GwtStatusServiceImpl.getNetworkStatus(GwtStatusServiceImpl.java:322)
	at org.eclipse.kura.web.server.GwtStatusServiceImpl.getDeviceConfig(GwtStatusServiceImpl.java:91)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
	... 53 more
```
</details>

**After this PR**:

<img width="881" alt="image" src="https://user-images.githubusercontent.com/22748355/220969108-c59e8850-2c6a-4d59-a3a0-d30de6561d6b.png">

--- a/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/status/wifi/WifiInterfaceStatus.java
+++ b/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/status/wifi/WifiInterfaceStatus.java
@@ -12,7 +12,6 @@
  ******************************************************************************/
 package org.eclipse.kura.net.status.wifi;
 
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.EnumSet;
 import java.util.List;

--- a/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/status/wifi/WifiInterfaceStatus.java
+++ b/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/status/wifi/WifiInterfaceStatus.java
@@ -98,10 +98,10 @@ public class WifiInterfaceStatus extends NetworkInterfaceStatus {
     public static class WifiInterfaceStatusBuilder extends NetworkInterfaceStatusBuilder<WifiInterfaceStatusBuilder> {
 
         private Set<WifiCapability> capabilities = EnumSet.of(WifiCapability.NONE);
-        private List<Long> supportedBitrates = Arrays.asList(0L);
+        private List<Long> supportedBitrates = Collections.emptyList();
         private Set<WifiRadioMode> supportedRadioModes = EnumSet.of(WifiRadioMode.UNKNOWN);
-        private List<Integer> supportedChannels = Arrays.asList(0);
-        private List<Long> supportedFrequencies = Arrays.asList(0L);
+        private List<Integer> supportedChannels = Collections.emptyList();
+        private List<Long> supportedFrequencies = Collections.emptyList();
         private String countryCode = "00";
         private WifiMode mode = WifiMode.UNKNOWN;
         private Optional<WifiAccessPoint> currentWifiAccessPoint = Optional.empty();

--- a/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java
+++ b/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java
@@ -16,6 +16,7 @@ package org.eclipse.kura.web.client.ui.network;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
 import java.util.logging.Logger;
 
 import org.eclipse.kura.web.client.messages.Messages;
@@ -415,13 +416,15 @@ public class TabWirelessUi extends Composite implements NetworkTab {
             this.selectedNetIfConfig = (GwtWifiNetInterfaceConfig) config;
             logger.severe("config: " + selectedNetIfConfig);
             this.activeConfig = this.selectedNetIfConfig.getActiveWifiConfig();
-            logger.severe("active config: " + this.activeConfig);
-            logger.severe("wireless mode: " + this.activeConfig.getWirelessMode());
-            logger.severe("channels " + this.activeConfig.getChannels());
-            if (this.activeConfig.getChannels() != null && !TabWirelessUi.this.activeConfig.getChannels().isEmpty()) {
-                updateChanneList(TabWirelessUi.this.activeConfig);
+            if (Objects.nonNull(this.activeConfig)) {
+                logger.severe("active config: " + this.activeConfig);
+                logger.severe("wireless mode: " + this.activeConfig.getWirelessMode());
+                logger.severe("channels " + this.activeConfig.getChannels());
+                if (this.activeConfig.getChannels() != null
+                        && !TabWirelessUi.this.activeConfig.getChannels().isEmpty()) {
+                    updateChanneList(TabWirelessUi.this.activeConfig);
+                }
             }
-
             loadCountryCode();
         }
 

--- a/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/GwtNetworkServiceImpl.java
+++ b/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/GwtNetworkServiceImpl.java
@@ -61,7 +61,7 @@ public class GwtNetworkServiceImpl {
                 } else {
                     Optional<NetworkInterfaceType> ifType = status.getNetInterfaceType(ifName);
                     if (ifType.isPresent()) {
-                        GwtNetInterfaceConfig gwtConfig = configuration.createGwtNetInterfaceConfig(ifName,
+                        GwtNetInterfaceConfig gwtConfig = configuration.getDefaultGwtNetInterfaceConfig(ifName,
                                 ifType.get());
                         status.fillWithStatusProperties(ifName, gwtConfig);
                         result.add(gwtConfig);

--- a/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/GwtNetInterfaceConfigBuilder.java
+++ b/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/GwtNetInterfaceConfigBuilder.java
@@ -101,14 +101,9 @@ public class GwtNetInterfaceConfigBuilder {
         this.gwtConfig.setName(this.ifName);
         this.gwtConfig.setHwName(this.ifName);
 
-        Optional<String> ifType = this.properties.getType(this.ifName);
-        if (ifType.isPresent()) {
-            this.gwtConfig.setHwType(ifType.get());
-        }
-
-        if (this.gwtConfig instanceof GwtWifiNetInterfaceConfig) {
-            String wifiMode = EnumsParser.getGwtWifiWirelessMode(this.properties.getWifiMode(this.ifName));
-            ((GwtWifiNetInterfaceConfig) gwtConfig).setWirelessMode(wifiMode);
+        Optional<String> interfaceType = this.properties.getType(this.ifName);
+        if (interfaceType.isPresent()) {
+            this.gwtConfig.setHwType(interfaceType.get());
         }
     }
 
@@ -163,6 +158,7 @@ public class GwtNetInterfaceConfigBuilder {
     private void setWifiProperties() {
         if (this.gwtConfig instanceof GwtWifiNetInterfaceConfig) {
             String wifiMode = EnumsParser.getGwtWifiWirelessMode(this.properties.getWifiMode(this.ifName));
+            ((GwtWifiNetInterfaceConfig) gwtConfig).setWirelessMode(wifiMode);
 
             if (wifiMode.equals(GwtWifiWirelessMode.netWifiWirelessModeAccessPoint.name())) {
                 setWifiMasterProperties();
@@ -183,7 +179,6 @@ public class GwtNetInterfaceConfigBuilder {
         gwtWifiConfig.setDriver(this.properties.getWifiMasterDriver(this.ifName));
         gwtWifiConfig.setIgnoreSSID(this.properties.getWifiMasterIgnoreSsid(this.ifName));
         gwtWifiConfig.setPassword(new String(this.properties.getWifiMasterPassphrase(this.ifName).getPassword()));
-        gwtWifiConfig.setChannels(this.properties.getWifiMasterChannel(this.ifName));
         gwtWifiConfig.setWirelessMode(
                 EnumsParser.getGwtWifiWirelessMode(this.properties.getWifiMasterMode(this.ifName)));
         gwtWifiConfig.setSecurity(
@@ -217,7 +212,6 @@ public class GwtNetInterfaceConfigBuilder {
         gwtWifiConfig.setDriver(this.properties.getWifiInfraDriver(this.ifName));
         gwtWifiConfig.setIgnoreSSID(this.properties.getWifiInfraIgnoreSsid(this.ifName));
         gwtWifiConfig.setPassword(new String(this.properties.getWifiInfraPassphrase(this.ifName).getPassword()));
-        gwtWifiConfig.setChannels(this.properties.getWifiInfraChannel(this.ifName));
         gwtWifiConfig
                 .setWirelessMode(EnumsParser.getGwtWifiWirelessMode(this.properties.getWifiInfraMode(this.ifName)));
         gwtWifiConfig

--- a/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServiceAdapter.java
+++ b/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServiceAdapter.java
@@ -64,14 +64,14 @@ public class NetworkConfigurationServiceAdapter {
     }
 
     /**
-     * Return an empty {@link GwtNetInterfaceConfig} for the given network interface
-     * name and type
+     * Return a {@link GwtNetInterfaceConfig} for the given network interface
+     * name and type with default values
      * 
      * @param ifName the network interface name
      * @param ifType the network interface type
      * @return a new {@link GwtNetInterfaceConfig}
      */
-    public GwtNetInterfaceConfig createGwtNetInterfaceConfig(String ifName, NetworkInterfaceType ifType) {
+    public GwtNetInterfaceConfig getDefaultGwtNetInterfaceConfig(String ifName, NetworkInterfaceType ifType) {
         GwtNetInterfaceConfig gwtConfig = new GwtNetInterfaceConfigBuilder()
                 .forInterface(ifName).forType(ifType).build();
         logger.debug("Created GWT Network Configuration for interface {} and type {}", ifName, ifType);

--- a/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java
+++ b/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java
@@ -205,6 +205,14 @@ public class NetworkStatusServiceAdapter {
                 wifiInterfaceInfo.getActiveWifiAccessPoint()
                         .ifPresent(accessPoint -> rssi.set(String.valueOf(accessPoint.getSignalQuality())));
                 gwtWifiNetInterfaceConfig.setHwRssi(rssi.get());
+                if (Objects.nonNull(gwtWifiNetInterfaceConfig.getStationWifiConfig())) {
+                    gwtWifiNetInterfaceConfig.getStationWifiConfig()
+                            .setChannels(wifiInterfaceInfo.getSupportedChannels());
+                } else {
+                    GwtWifiConfig gwtConfig = new GwtWifiConfig();
+                    gwtConfig.setChannels(wifiInterfaceInfo.getSupportedChannels());
+                    gwtWifiNetInterfaceConfig.setStationWifiConfig(gwtConfig);
+                }
             }
         }
     }

--- a/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/utils/EnumsParser.java
+++ b/kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/utils/EnumsParser.java
@@ -60,7 +60,7 @@ public class EnumsParser {
             }
         }
 
-        return GwtWifiWirelessMode.netWifiWirelessModeDisabled.name();
+        return GwtWifiWirelessMode.netWifiWirelessModeStation.name();
     }
 
     /**
@@ -166,7 +166,7 @@ public class EnumsParser {
             }
         }
 
-        return GwtWifiSecurity.netWifiSecurityNONE.name();
+        return GwtWifiSecurity.netWifiSecurityWPA2.name();
     }
 
     /**
@@ -214,7 +214,7 @@ public class EnumsParser {
             }
         }
 
-        return GwtWifiCiphers.netWifiCiphers_NONE.name();
+        return GwtWifiCiphers.netWifiCiphers_CCMP.name();
     }
 
     /**
@@ -268,7 +268,8 @@ public class EnumsParser {
             }
         }
 
-        return Optional.empty();
+        return Optional.of(GwtWifiRadioMode.netWifiRadioModeBGN.name());
+
     }
 
     /**

